blocks/recentreports/recentreports.js

```diff
@@ -0,0 +1,253 @@
+import {
+  buildBlock, decorateBlock, loadBlock, readBlockConfig,
+} from '../../scripts/aem.js';
+import {
+  createElement, observe, getResearchAPIUrl, getDataFromAPI,
+} from '../../scripts/blocks-utils.js';
+
+function decorateBoxHeader(title, reportLink) {
+  const heading = createElement('h3', '');
+  const anchor = createElement('a', '');
+  anchor.href = reportLink;
+  anchor.target = '_blank';
+  anchor.tabIndex = 0;
+  anchor.textContent = title;
+  heading.appendChild(anchor);
+  return heading;
+}
+
+/**
+ * Formats a date string into a human-readable format.
+ * @param {string} dateString - The date string to be formatted.
+ * @returns {string} The formatted date string.
+ */
+function formatDate(dateString) {
+  if (dateString) {
+    const date = new Date(dateString);
+    const day = date.getDate().toString().padStart(2, '0');
+    const month = new Intl.DateTimeFormat('en-US', { month: 'short' }).format(date);
+    const year = date.getFullYear();
+    const hour = date.getHours().toString().padStart(2, '0');
+    const minute = date.getMinutes().toString().padStart(2, '0');
+
+    return `${day} ${month} ${year} | ${hour}:${minute}`;
+  }
+  return '';
+}
+
+function formatPriceInRupees(price) {
+  // Check for empty or null input
+  if (price === null || price === '') {
```

Comment:
we can also have an alternative instead
`if (!price) {
    return '';
  }`

Reply:
This would show blank when price is 0. ).00 looks better on the card.